websites/T/TikTok/presence.ts

```diff
@@ -18,101 +19,142 @@ async function getStrings() {
 	);
 }
 
-let strings: Awaited<ReturnType<typeof getStrings>>,
-	oldLang: string = null;
+const newStrings = getStrings();
+let strings: Awaited<typeof newStrings>;
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
 			largeImageKey: "tiktok",
 			startTimestamp: browsingTimestamp
 		},
 		newLang = await presence.getSetting<string>("lang").catch(() => "en"),
-		[, page, pageType] = location.pathname.split("/");
-
-	if (oldLang !== newLang || !strings) {
-		oldLang = newLang;
-		strings = await getStrings();
-	}
+		[, page, pageType] = location.pathname.split("/"),
+		[buttons, covers] = await Promise.all([
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("covers")
+		]);
+	strings ??= await newStrings;
```

Comment:
Are you sure this will get updated

Reply:
yes

Reply:
Well, I'm not sure if you misunderstood me. But it won't get updated if you change a language. So just revert this change.